exporter/tanzuobservabilityexporter/transformer.go

```diff
@@ -86,12 +88,30 @@ func (t *traceTransformer) Span(orig pdata.Span) (span, error) {
 		SpanID:         spanID,
 		ParentSpanID:   parentSpanIDtoUUID(orig.ParentSpanID()),
 		Tags:           tags,
+		Source:         source,
 		StartMillis:    startMillis,
 		DurationMillis: durationMillis,
 		SpanLogs:       eventsToLogs(orig.Events()),
 	}, nil
 }
 
+func getSource(attributes pdata.AttributeMap) string {
+	candidateKeys := []string{"source", conventions.AttributeHostName, "hostname", conventions.AttributeHostID}
+
+	var source string
+
+	for _, key := range candidateKeys {
+		if value, isFound := attributes.Get(key); isFound {
+			source = value.StringVal()
+			attributes.Delete(key)
```

Comment:
Looking at the [Data Ownership](https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter#data-ownership) section from the Collector's Exporter docs, we shouldn't be modifying any of the data in `pdata.Traces`. So, we should not call `attributes.Delete()`. Let's find another way to do this, such as by returning the source and resources attributes as tags. The signature of `getSource` might change to:
```go
func getSourceAndResourceTags(attributes pdata.AttributeMap) (string, map[string]string)
```

Then we combine the resource tags with the span tags.

Reply:
Creating a copy before we remove.